pkgs/servers/web-apps/whitebophir/default.nix

```
@@ -0,0 +1,60 @@
+{ stdenv
+, pkgs
+, lib
+, fetchurl
+, nodejs
+}:
+
+let
+  nodePackages = import ./node-packages.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+
+  # find an element in an attribute set
+  findValue = pred: default: set:
+    let
+      list =
+        lib.concatMap
+        (name:
+          let v = set.${name}; in
+          if pred name v then [v] else []
+        )
+        (lib.attrNames set)
+        ;
+    in
+      if list == [] then default
+      else lib.head list
+      ;
+
+  # The whitebophir package attribute key changes for each release. Get it out
+  # programatically instead.
+  whitebophir = findValue
+    (k: v: v.packageName == "whitebophir")
+    (throw "whitebophir not found")
+    nodePackages
+    ;
+
+  combined = whitebophir.override {
+    postInstall = ''
+      out_whitebophir=$out/lib/node_modules/whitebophir
+
+      # add executable
+      mkdir $out/bin
+      cat <<EOF > $out/bin/whitebophir
+      #!${stdenv.shell}
+      exec ${nodejs}/bin/node $out_whitebophir/server/server.js
+      EOF
+      chmod +x $out/bin/whitebophir
+    '';
+
+    meta = with stdenv.lib; {
```

```suggestion
    meta = with lib; {
```

See https://github.com/NixOS/nixpkgs/issues/108938